src/App.tsx

```diff
@@ -1,61 +1,156 @@
-import './App.scss';
+// import './App.scss';
+import 'bulma';
+import { useState } from 'react';
+import { TodoList } from './components/TodoList';
+import { getPreparedData } from './helpers/getPreparedData';
+import { getUser } from './helpers/getUser';
+import { getId } from './helpers/getNewTodoId';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoWithUser } from './Types';
+
+const preparedData = getPreparedData(
+  todosFromServer,
+  usersFromServer,
+);
 
 export const App = () => {
+  const [preparedDodos, setPreparedTodos] = useState(preparedData);
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [titleQueue, setTitleQueue] = useState('');
+  const [isFieldsEmpty, setIsFieldsEmpty] = useState(false);
+
+  const addNewTodo = (
+    userId: number,
+    title: string,
+  ) => {
+    const newTodo: TodoWithUser = {
+      id: getId(todosFromServer),
+      title,
+      completed: false,
+      userId,
+      user: getUser(userId, usersFromServer),
+    };
+
+    setPreparedTodos((currentTodos) => ([
+      ...currentTodos,
+      newTodo,
+    ]));
+  };
+
+  const clearForm = () => {
+    setSelectedUser(0);
+    setTitleQueue('');
+  };
+
+  const handleFormSubmit = (
+    event: React.FormEvent,
+  ) => {
+    event.preventDefault();
+    const isInputsValid = selectedUser && titleQueue;
+
+    if (isInputsValid) {
+      clearForm();
+      setIsFieldsEmpty(false);
+      addNewTodo(selectedUser, titleQueue);
+    } else {
+      setIsFieldsEmpty(true);
+    }
```

Comment:
Try to invert the condition, I think code will look more readable
```suggestion
    if (!isInputsValid) {
      return setIsFieldsEmpty(true);
    } 
    
      clearForm();
      setIsFieldsEmpty(false);
      addNewTodo(selectedUser, titleQueue);
```